components/automate-gateway/gateway/middleware/authv2/authv2.go

```diff
@@ -83,13 +83,44 @@ func (c *client) Handle(ctx context.Context, subjects []string, projectsToFilter
 	return auth_context.NewContext(ctx, subjects, projects, resource, action, middleware.AuthV2.String()), nil
 }
 
+type resp struct {
+	ctx        context.Context
+	authorized bool
+}
+
+
+func (r *resp) Ctx() context.Context {
+	return r.ctx
+}
+
+func (r *resp) GetAuthorized() bool {
+	return r.authorized
+}
+
 func (c *client) IsAuthorized(ctx context.Context, subjects []string, resource, action string,
-) (middleware.AuthorizationResponse, error) {
-	return c.client.IsAuthorized(ctx, &authz.IsAuthorizedReq{
-		Subjects: subjects,
-		Resource: resource,
-		Action:   action,
+	projectsToFilter []string) (middleware.AuthorizationResponse, error) {
+	log := ctxlogrus.Extract(ctx)
+	filteredResp, err := c.client.ProjectsAuthorized(ctx, &authz.ProjectsAuthorizedReq{
+		Subjects:       subjects,
+		Resource:       resource,
+		Action:         action,
+		ProjectsFilter: projectsToFilter,
 	})
+	if err != nil {
+		if status.Convert(err).Code() == codes.FailedPrecondition {
+			return nil, err
+		}
+		log.WithError(err).Error("error authorizing request")
```

Comment:
I'm curious about the choice to log here-- can we not guarantee that the error itself will surface in the logs? 

If that's the case, I imagine it would be helpful to have the subj/resource/etc in the log message as well as the error.

Reply:
😄 a) I don't think I've spent much time deliberating the choice of log here.
b) *ANY* GRPC response that has a status !=`OK` will be logged by the gateway. So, we should end up having [most of the stuff in the logs because of the status message below](https://github.com/chef/automate/pull/380/files#diff-cecefbe2195758a5155eeb1d419ed6cfR115).

Good enough? 😉